control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3204,3 +3213,55 @@ func (r *HostedControlPlaneReconciler) reconcileCSISnapshotControllerOperator(ct
 
 	return nil
 }
+
+func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControlPlane) {
+	if hcp.Spec.Platform.AWS == nil {
+		return
+	}
+
+	log := ctrl.LoggerFrom(ctx)
+
+	awsSession := awsutil.NewSession("control-plane-operator", "", "", "", "")
+	awsConfig := awssdk.NewConfig()
+	awsConfig.Region = awssdk.String("us-east-1")
+	route53Client := route53.New(awsSession, awsConfig)
+
+	// We try to interact with cloud provider to see validate is operational.
+	if _, err := route53Client.ListHostedZonesByNameWithContext(ctx, &route53.ListHostedZonesByNameInput{
+		DNSName: pointer.String(fmt.Sprintf("%s.%s", hcp.Name, hypershiftLocalZone)),
+	}); err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			// When awsErr.Code() is WebIdentityErr it's likely to be an external issue, e.g the idp resource was deleted.
```

Comment:
Should you check for WebIdentityErr as the error code to report that the identity provider is unavailable? The call could fail for a lot of other reasons.

Reply:
do you want WebIdentityErr to be a "Reason" and keep capturing the other errors as AWSErrorReason, or do you want to ignore them?

Reply:
I think it would be more correct to say Status = Unknown and set the reason to AWSErrorReason

Reply:
updated to have its own reason so consumers can react programatically